internal/tcpchecker/portchecker.go

```diff
@@ -0,0 +1,27 @@
+package tcpchecker
+
+import (
+	"time"
+
+	"github.com/lunarway/strong-duckling/internal/daemon"
+	"github.com/prometheus/common/log"
+)
+
+type PortChecker struct {
+	address string
+	port    int
+}
+
+func StartPortChecking(address string, port int, reporter Reporter) *daemon.Daemon {
```

Comment:
Maybe this should be renamed to something like `New*` now that it does not actually start anything?
```suggestion
func NewPortCheckerDaemon(address string, port int, reporter Reporter) *daemon.Daemon {
```

Or as the package is named `tcpchecker` we could reduce it to:
```suggestion
func NewDaemon(address string, port int, reporter Reporter) *daemon.Daemon {
```

Usage would then be:
```diff
- tcpcheker.StartPortChecking(address, port, reporter)
+ tcpchecker.NewDaemon(address, port, reporter)
```

I'm unsure if the package would ever allow a non-port checker? As is the reason to omit the `port` section of the name.

Reply:
👍